WeakAurasOptions/OptionsFrames/TextEditor.lua

```diff
@@ -193,106 +304,346 @@ local function ConstructTextEditor(frame)
+      frame:AddChild(button)
+    end
 
-  -- CTRL + S saves and closes, ESC cancels and closes
-  editor.editBox:HookScript("OnKeyDown", function(_, key)
-    if IsControlKeyDown() and key == "S" then
-      group:Close()
+    local heading2 = AceGUI:Create("Heading")
+    heading2:SetText(L["Your Saved Snippets"])
+    heading2:SetRelativeWidth(1)
+    frame:AddChild(heading2)
+
+    -- iterate saved snippets and make buttons
+    for order, snippet in ipairs(savedSnippets) do
+      local button = AceGUI:Create("WeakAurasSnippetButton")
+      button:SetTitle(snippet.name)
+      button:SetDescription(snippet.snippet)
+      button:SetEditable(true)
+      button:SetRelativeWidth(1)
+      button:SetCallback(
+        "OnClick",
+        function()
+          WeakAuras.editor.editBox:Insert(snippet.snippet)
+          WeakAuras.editor:SetFocus()
+        end
+      )
+      button.deleteButton:SetScript(
+        "OnClick",
+        function()
+          table.remove(savedSnippets, order)
+          UpdateSnippets(frame)
+        end
+      )
+      button:SetCallback(
+        "OnEnterPressed",
+        function()
+          local newName = button.renameEditBox:GetText()
+          if newName and #newName > 0 then
+            local found = false
+            for _, snippet in ipairs(savedSnippets) do
+              if snippet.name == newName then
+                found = true
+                break
+              end
+            end
+            if not found then
+              savedSnippets[order].name = newName
+              UpdateSnippets(frame)
+            end
+          end
+        end
+      )
+      frame:AddChild(button)
     end
-    if key == "ESCAPE" then
-      group:CancelClose()
+  end
+
+  -- Make sidebar for snippets
+  local snippetsFrame = CreateFrame("FRAME", "WeakAurasSnippets", group.frame)
+  snippetsFrame:SetPoint("TOPLEFT", group.frame, "TOPRIGHT", 20, 0)
+  snippetsFrame:SetPoint("BOTTOMLEFT", group.frame, "BOTTOMRIGHT", 20, 0)
+  snippetsFrame:SetWidth(250)
+  snippetsFrame:SetBackdrop(
+    {
+      bgFile = "Interface\\DialogFrame\\UI-DialogBox-Background",
+      edgeFile = "Interface\\DialogFrame\\UI-DialogBox-Border",
+      tile = true,
+      tileSize = 32,
+      edgeSize = 32,
+      insets = {left = 8, right = 8, top = 8, bottom = 8}
+    }
+  )
+  snippetsFrame:SetBackdropColor(0, 0, 0, 1)
+
+  -- Add button to save new snippet
+  local AddSnippetTextBox = CreateFrame("EditBox", nil, snippetsFrame, "InputBoxTemplate")
+  AddSnippetTextBox:SetPoint("TOPLEFT", snippetsFrame, "TOPLEFT", 25, -15)
+  AddSnippetTextBox:SetHeight(20)
+  AddSnippetTextBox:SetWidth(100)
+  AddSnippetTextBox:SetAutoFocus(false)
+  local AddSnippetButton = CreateFrame("Button", nil, snippetsFrame, "UIPanelButtonTemplate")
+  AddSnippetButton:SetPoint("TOPLEFT", AddSnippetTextBox, "TOPRIGHT")
+  AddSnippetButton:SetHeight(20)
+  AddSnippetButton:SetWidth(100)
+  AddSnippetButton:SetText(L["Add Snippet"])
```

Comment:
The german translation would be: "Snippet hinzufügen", which doesn't fit into that space.

You can either:
* Rename it to "Add", which is "Hinzufügen" in german and might fit.
* Use the same workflow as the main window. That is create a new aura doesn't ask for a name. Instead the aura is just created but with the editbox being focussed and editable.
* Move the button to the main window, as a "Save Snippet" button.